crates/recent_projects/src/recent_projects.rs

```diff
@@ -294,7 +296,7 @@ impl PickerDelegate for RecentProjectsDelegate {
                         match candidate_workspace_location {
                             SerializedWorkspaceLocation::Local(paths, _) => {
                                 let paths = paths.paths().as_ref().clone();
-                                if replace_current_window {
+                                if replace_current_window || WorkspaceSettings::get_global(cx).open_new_workspace_in_current_window {
```

Comment:
One thing I'm not sure about is if this is correct, feel like we have to remove it here and replace `false` on line 318 with this again `WorkspaceSettings::get_global(cx).open_new_workspace_in_current_window`. But maybe one of the Zed team members can correct me here.

Reply:
Not sure about that. Then `prepare_to_close` call will be skipped. But I don't know if it is required or no.

Reply:
Looked at the code, I think what you have is good. Because if you want to replace the workspace you also have to hang up the call. And that is what that code handles.